app/models/request.rb

```diff
@@ -0,0 +1,73 @@
+class Request < ActiveRecord::Base
+  belongs_to :requestee, :class_name => "User"
+  belongs_to :requester, :class_name => "User"
+
+  validates_presence_of :requestee
+  validates_presence_of :requester
+  validates_presence_of :item_type
+  validates_inclusion_of :item_type, in: ["Movie", "AndroidApp"]
+  validates_presence_of :status
+  validates_inclusion_of :status, in: ["pending", "sent", "seen", "successful"]
+
+  validates_uniqueness_of :item_type, scope: [:requestee, :requester],
```

Comment:
If a request is `pending` or `sent` but not `seen` it implies the `requestee` hasn't yet acknowledged the request. So the other guy shouldn't be allowed to make more requests till then. Will probably need to check if the request is too old as well. If so, it should be allowed.